### PR TITLE
fix(Icon): make "name" prop required

### PR DESCRIPTION
**Pull request by lucivpav**
An instantiated `Icon` without a `name` makes no sense. This PR makes the `name` prop mandatory. An alternative would be to set a default name value. This resolves issue #1715.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - `Loader` uses `Text` component for `label` slot instead of `Box` @layershifter ([#1701](https://github.com/stardust-ui/react/pull/1701))
 - Fix test cut off in `Button` component @layershifter ([#1716](https://github.com/stardust-ui/react/pull/1716))
 - Close `Toolbar`'s menu when it looses focus @sophieH29 ([#1688](https://github.com/stardust-ui/react/pull/1688))
+- Fix `Icon` `name` prop not required @lucivpav ([#1723](https://github.com/stardust-ui/react/pull/1723))
```
Comment (layershifter) [inline CHANGELOG.md]:
```suggestion
- Require `name` prop in `Icon` component @lucivpav ([#1723](https://github.com/stardust-ui/react/pull/1723))
```

This like a bit cleaner to me

Comment (layershifter) [reply CHANGELOG.md]: Please also update `propTypes`:

```diff
-name: PropTypes.string,
+name: PropTypes.string.isRequired,
```

**Review by layershifter — APPROVED** [review]

File: packages/react/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
   disabled?: boolean
 
   /** Name of the icon. */
-  name?: string
+  name: string
```
Comment (mnajdova) [inline packages/react/src/components/Icon/Icon.tsx:33]: propTypes too?

Comment (lucivpav) [reply packages/react/src/components/Icon/Icon.tsx:33]: Fixed just now

**Comment by lucivpav** [comment]
@layershifter can you please take a look at the new changes? I have fixed the test errors.

**Pull request merged**